### PR TITLE
tip-manager project

### PR DESCRIPTION
**Pull request by vc-jinal**
- CRUD APIs for User Management.
- User Authentication using JWT.
- Encrypt Password.
- List of Tips calculated by user with particular place, total amount paid and the time.
- Most selected percentage of tips like what was the repeated Tip Percentage.
- Most visited places.

### CODE REVIEW

File: tip-manager/db/connection.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+const { MongoClient } = require('mongodb');
+
+//connection url
+const url = 'mongodb://127.0.0.1:27017';
```
Comment (vc-mohit) [inline tip-manager/db/connection.js:4]: it should be in env

File: tip-manager/route/place.route.js
Comment (vc-mohit) [inline tip-manager/route/place.route.js:1]: can we create a common function for the highest occurrence in an array?

File: tip-manager/route/place.route.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        return res.send({ statusCode: 404, message: "user not found" })
+    }
+
+    const addPlaceDetails = {
+        user_id: new ObjectId(req.user.id),
+        placeName: placeName,
+        billAmount: billAmount,
+        tipAmount: tipAmount,
+        totalAmount: billAmount + tipAmount,
+        tipPercentage: Math.round(tipAmount / billAmount * 100),
```
Comment (vc-mohit) [inline tip-manager/route/place.route.js:25]: get upto 1 decimal value

File: tip-manager/db/connection.js (hunk 1 above)
Comment (vc-jinal) [reply tip-manager/db/connection.js:4]: changed connection URL to env

File: tip-manager/route/place.route.js
Comment (vc-jinal) [reply tip-manager/route/place.route.js:1]: created common function

File: tip-manager/route/place.route.js (hunk 2 above)
Comment (vc-jinal) [reply tip-manager/route/place.route.js]: fixed this

**Review by vc-mohit — APPROVED** [review]

**Review by vc-mohit — APPROVED** [review]

**Pull request merged**